lib/NUPACK/emscripten/FullEnsemble.cpp

```diff
@@ -209,38 +210,81 @@ FullAdvancedResult* FullEnsembleNoBindingSite(const std::string& seqString, int
 
     clearDnaStructures(&suboptStructs);
 
-    //get defect here later for now popuolate with 0
-    //get defect here later for now popuolate with 0
-    auto autoStructString = MakeCString(mfeStructure);
-    char* newSecondStruct = autoStructString.get();
-
-    //get ensemble defect for mfe structure
-     Defect ensembleDefectStruct;
-     double ensembleDefect = -2;
-     double ensembleDefectNormalized = -3;
-     double mfeDefect = -4;
-     double mfeDefectNormalized = -5;
-     bool getDefectMFE = false;
-    ensembleDefectStruct = getEnsembleDefect(string, newSecondStruct, temperature, pseudoknotted, false, getDefectMFE);
-    ensembleDefect = ensembleDefectStruct.ensembleDefect;
-    ensembleDefectNormalized = ensembleDefectStruct.ensembleDefectNormalized;
-    
-    getDefectMFE=true;
-    ensembleDefectStruct = getEnsembleDefect(string, newSecondStruct, temperature, pseudoknotted, false, getDefectMFE);
-    mfeDefect = ensembleDefectStruct.mfeDefect;
-    mfeDefectNormalized = ensembleDefectStruct.mfeDefectNormalized;
+    return result;
+}
 
-    result->ensembleDefect=ensembleDefect;
-    result->ensembleDefectNormalized=ensembleDefectNormalized;
-    result->mfeDefect=mfeDefect;
-    result->mfeDefectNormalized=mfeDefectNormalized;
+/*
+  mode is how to do the ED
+  1= do only ED
+  2 = do only MFE ED
+  3 = do both
+*/
 
+FullDefectResult* FullEnsembleDefect( const std::string& seqString, const std::string& MfeStructure, int temperature, bool const pseudoknotted = false, int const mode = 1) {
+
+  auto autoSeqString = MakeCString(seqString);
+  char* string = autoSeqString.get();
+
+  auto autoMfeStructure = MakeCString(MfeStructure);
+  char* MfeStructureString = autoMfeStructure.get();
+
+  
+
+  //bool array for 
+  bool testList[2] = {false};
```

Comment:
Why is this a bool array instead of just two variables (with names that clearly correspond to their purposes)?

Reply:
I think I was playing around with an idea and when it worked I forgot to change the names although I think I meant testList as in a list of thigs to test... but I see the confusion and will get changed

Reply:
i made this two variables and got rid of extra code